modules/operator-roles/main.tf

```diff
@@ -0,0 +1,113 @@
+      service_accounts   = ["system:serviceaccount:kube-system:kube-controller-manager"]
+    },
+    {
+      operator_name      = "capa-controller-manager"
+      operator_namespace = "kube-system"
+      role_name          = "kube-system-capa-controller-manager"
+      policy_details     = "arn:aws:iam::aws:policy/service-role/ROSANodePoolManagementPolicy"
+      service_accounts   = ["system:serviceaccount:kube-system:capa-controller-manager"]
+    },
+    {
+      operator_name      = "control-plane-operator"
+      operator_namespace = "kube-system"
+      role_name          = "kube-system-control-plane-operator"
+      policy_details     = "arn:aws:iam::aws:policy/service-role/ROSAControlPlaneOperatorPolicy"
+      service_accounts   = ["system:serviceaccount:kube-system:control-plane-operator"]
+    },
+    {
+      operator_name      = "kms-provider"
+      operator_namespace = "kube-system"
+      role_name          = "kube-system-kms-provider"
+      policy_details     = "arn:aws:iam::aws:policy/service-role/ROSAKMSProviderPolicy"
+      service_accounts   = ["system:serviceaccount:kube-system:kms-provider"]
+    },
+  ]
+  operator_roles_count = length(local.operator_roles_properties)
+  operator_role_prefix = var.operator_role_prefix
+  path                 = coalesce(var.path, "/")
+}
+
+data "aws_iam_policy_document" "custom_trust_policy" {
+  count = local.operator_roles_count
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    principals {
+      type        = "Federated"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${var.oidc_endpoint_url}"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "${var.oidc_endpoint_url}:sub"
+      values   = local.operator_roles_properties[count.index].service_accounts
+    }
+  }
+}
+
+resource "aws_iam_role" "operator_role" {
+  count                = local.operator_roles_count
+  name                 = substr("${local.operator_role_prefix}-${local.operator_roles_properties[count.index].operator_namespace}-${local.operator_roles_properties[count.index].operator_name}", 0, 64)
+  permissions_boundary = var.permissions_boundary
+  path                 = local.path
+  assume_role_policy   = data.aws_iam_policy_document.custom_trust_policy[count.index].json
+
+  tags = merge(var.tags, {
+    rosa_managed_policies = true
+    rosa_hcp_policies     = true
+    red-hat-managed       = true
+    operator_namespace    = local.operator_roles_properties[count.index].operator_namespace
+    operator_name         = local.operator_roles_properties[count.index].operator_name
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "operator_role_policy_attachment" {
+  count      = local.operator_roles_count
+  role       = aws_iam_role.operator_role[count.index].name
+  policy_arn = local.operator_roles_properties[count.index].policy_details
```

Comment:
The policies used here are managed policies, right?
What about the shared VPC scenario? how do you make sure to use the right policy in this case?

Reply:
There's no shared vpc yet in HCP